crates/cache/src/lib.rs

```diff
@@ -94,20 +96,47 @@ impl QueryResultCacheProvider {
     ///
     /// Will return `Err` if method fails to access the cache
     pub async fn get(&self, plan: &LogicalPlan) -> Result<Option<CachedQueryResult>> {
-        self.cache.get(plan).await
+        metrics::counter!("results_cache_request_count").increment(1);
+        match self.cache.get(plan).await {
+            Ok(Some(cached_result)) => {
+                metrics::counter!("results_cache_hit_count").increment(1);
+                Ok(Some(cached_result))
+            }
+            Ok(None) => Ok(None),
+            Err(e) => Err(e),
+        }
     }
 
     /// # Errors
     ///
     /// Will return `Err` if method fails to access the cache
     pub async fn put(&self, plan: &LogicalPlan, result: CachedQueryResult) -> Result<()> {
-        self.cache.put(plan, result).await
+        let res = self.cache.put(plan, result).await;
+
+        #[allow(clippy::cast_precision_loss)]
+        metrics::gauge!("results_cache_max_size").set(self.max_size() as f64);
```

Comment:
`results_cache_max_size` is constant value but periodically reporting this metric is useful for charts/etc

Reply:
Even if we report it once, it will still show up in the Prometheus metrics every time - so reporting it multiple times doesn't really buy us anything. I would move this to the `QueryResultCacheProvider::new()` function.

Reply:
Moved to `QueryResultCacheProvider::new()` - thank you!